### ISSUE TITLE
Cannot read YAML !!set

### ISSUE THREAD

**Issue by jmcgeheeiv** [mzdaniel/loadconfig#6]
You defined `loadconfig.ODict.load.Loader.construct_mapping()` to construct an ODict from YAML.

This, however shadows the original `construct_mapping()`.  When `yaml.constructor.SafeConstructor.construct_yaml_set()` tries to use this, the signatures are different, so an `AttributeError` is raised.

The solution is simple.  Merely rename `loadconfig.ODict.load.Loader.construct_mapping()` to something unique, like `odict_mapping()`.

**Comment by mzdaniel** [comment]
Hi John. Glad you are trying loadconfig.
Not sure what is not working for you. Could you post the test you are trying?

**Comment by jmcgeheeiv** [comment]
Since I have fairly strict semantics, I am making my own "loadconfig", using just your ODict.  You are credited, of course.

Here is the unittest test that I used to find the problem described in this issue.  Specifically, the thing that ODict cannot handle is YAML `!!set {}`:

```
    def test_load_set(self):
        '''Load a YAML !!set

        Note that this test case has been modified to fit the ODict case without
        actually running ODict.
        '''
        yaml_string = '''
            app:
                option:
                    default: !!set {}
            '''
        expected_object = {'app': {'option': {'default': set()}}}
        safe_loaded = yaml.safe_load(yaml_string)
        self.assertEqual(safe_loaded, expected_object, "ordinary safe_load loads !!set")

        self.assertEqual(ODict(yaml_string), expected_object, "ODict loads !!set")
```

**Comment by mzdaniel** [comment]
I didn't considered sets in a config, but as you are bringing it up, here is a quick fix so you can get going until I have more time to understand why this is happening:

```
diff --git a/loadconfig/__init__.py b/loadconfig/__init__.py
index d51814b..147c359 100644
--- a/loadconfig/__init__.py
+++ b/loadconfig/__init__.py
@@ -164,7 +164,9 @@ class Odict(OrderedDict):
                         return ''
                 return data

-            def construct_mapping(self, safeloader, node):
+            def construct_mapping(self, *args):
+                node = args[-1]
+                safeloader = self if len(args) < 2 else args[0]
                 safeloader.flatten_mapping(node)
                 return Odict(safeloader.construct_pairs(node))
```

Did loadconfig tests passed with your solution?  Feel free to submit a pull request.

**Comment by jmcgeheeiv** [comment]
After quite a bit of tracing through the various `construct_mapping()` with the debugger, I am convinced that merely changing the name of your `construct_mapping()` is the correct solution.

**Comment by mzdaniel** [comment]
Yes, I reached the same conclusion too. I am preparing for a new release making YamlLoader its own class and some other adjustments.
Can you make a pull request to the dev branch? (it will automatically trigger loadconfig tests on buildbot.)
Do you mind posting a link of the project you are working? I wonder how are you using sets in your configs.

**Comment by mzdaniel** [comment]
Thank you @jmcgeheeiv for finding this loadconfig edge-case. I translated your unittest to pytest and credited you in the commit message that fix this use case.
